docs/source/developers/reviewing.rst

```diff
@@ -0,0 +1,226 @@
+  values, of course). Does a function allow different input types? etc.
+
+- If some aspect of a functionality is delicate (either by definition or
+  as an implementation detail), it should be tested.
+
+- Corner cases should be exercised, especially in low-level implementation
+  languages such as C++.  Examples: empty arrays, zero-chunk arrays, arrays
+  with only nulls, etc.
+
+- Stress tests can be useful, for example to uncover synchronizations bugs
+  if non-trivial parallelization is being added, or to validate a computational
+  argument against a slow and straightforward reference implementation.
+
+- A mitigating concern, however, is the overall cost of running the test
+  suite.  Continuous Integration (CI) runtimes can be painfully long and
+  we should be wary of increasing them too much.  Sometimes it is
+  worthwhile to fine-tune testing parameters to balance the usefulness
+  of tests against the cost of running them (especially where stress tests
+  are involved, since they tend to imply execution over large datasets).
+
+
+Social aspects
+==============
+
+* Reviewing is a communication between the contributor and the reviewer.
+  Avoid letting questions or comments unanswered for too long ("too long"
+  is of course very subjective, but two weeks can be a reasonable
+  heuristic).  If you cannot allocate time soon, do say it explicitly.
+  If you don't have the answer to a question, do say it explicitly.
+  Saying "I don't have time immediately but I will come back later,
+  feel free to ping if I seem to have forgotten" or "Sorry, I am out of depth
+  here" is always better than saying nothing and letting the interlocutor
+  wondering.
+
+* If you know someone who may help on a blocking issue and past experience
+  suggests they can be available for that, feel free to add them to the
+  discussion (for example by cc'ing their Github handle).
+
+* If the contributor has stopped giving feedback or updating their PR,
+  perhaps they're not interested anymore, but perhaps also they're stuck
+  on some issue and feel unable to push their contribution any further.
+  Don't hesitate to ask ("I see this PR hasn't seen any updates recently,
+  are you stuck on something? Do you need any help?").
+
+* If the contribution is genuinely desirable and the contributor is not making
+  any progress, it is also possible to take it up.  Out of politeness,
+  it is however better to ask the contributor first.
+
+* Some contributors are looking for a quick fix to a specific problem and
+  don't want to spend too much time on it.  Others on the contrary are eager
+  to learn and improve their contribution to make it conform to the
+  project's standards.  The latter kind of contributors are especially
+  valuable as they may become long-term contributors or even committers
+  to the project; it can therefore be a good strategy to prioritize
+  interactions with such contributors.
```

Comment:
I get the point here, but I worry this sounds a little exclusive; perhaps we can say instead that we should help people who are interested in getting further involved with the community to do so?

Reply:
Hmm, I may simply remove the last proposition ("it can therefore be a good strategy...").